editor/blocks/heading-block/index.js

```diff
@@ -0,0 +1,23 @@
+const { query, html, prop } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'wp/heading', {
+	title: 'Heading',
+	icon: 'heading',
+
+	attributes: {
+		value: query( 'h1,h2,h3,h4,h5,h6', html() ),
+		headingType: query( 'h1,h2,h3,h4,h5,h6', prop( 'nodeName' ) ),
+	},
+
+	edit( attributes, onChange ) {
+		return wp.element.createElement( wp.blocks.EditableHeading, {
```

Comment:
Noted on the other pull request too, but we should only need one `Editable` component which manages rich text edits. In this case you want to allow edits within the heading, so you'd render it as a child:

```jsx
const { headingType = 'H1', value } = attributes;
const Heading = headingType.toLowerCase();

return (
	<Heading>
		<Editable
			value={ value }
			onChange={ ( nextValue ) => onChange( { value: nextValue } ) } />
	</Heading>
);
```